### PR TITLE
Change query for find custom-url by page to improve performance

### LINKED ISSUE #6999: Content repository query on publish and on delete is very slow
| Q | A
| --- | ---
| Sulu Version | 2.5.5 with article bundle @2.4
| PHP Version | 8.1
| DB Version | mariadb 10.6
| Browser Version | all

# Actual Behavior

When publishing or deleting a document in a sulu application with 10000+ aritcles, the response is very slow depending on the server hardware. It might take up to 15s before getting a response.
With the help of the symfony dev toolbar I found a specific query that causes such long response times:
```
SELECT n0.path AS n0_path, n0.identifier AS n0_identifier, n0.props AS n0_props FROM phpcr_nodes n0 WHERE n0.workspace_name = ? AND n0.type IN ('nt:unstructured', 'rep:root') AND (EXTRACTVALUE(n0.props, 'count(//sv:property[@sv:name="jcr:mixinTypes"]/sv:value[text()="sulu:custom_url"]) > 0') AND EXTRACTVALUE(n0.props, 'count(//sv:property[@sv:name="sulu:target"]/sv:value[text()="53548267-f07e-46b2-a61f-093b63519f43"]) > 0'))
```
In average I've seen query times between 5 and 9 seconds.

# Expected Behavior

The query does not slow down the actual response.

# Steps to Reproduce

Have a sulu application with at least 10000 articles and try to publish or delete a document (e.g. a default page in the default webspace).

# Possible Solutions

At least on publish do not check every other document for references of the currently created document.

### PR DESCRIPTION
**Pull request by wachterjohannes**
| Q | A
| --- | ---
| Bug fix? | yes
| New feature? | no
| BC breaks? | no
| Deprecations? | no
| Fixed tickets | fixes #6999
| Related issues/PRs | none
| License | MIT
| Documentation PR | none

#### What's in this PR?

This PR improves an issue with a phpcr query for big content-repositories,

### CODE REVIEW

**Review by alexander-schranz — APPROVED** [review]

**Comment by wachterjohannes** [comment]
@rgericke-elb could you please test this PR and give us feedback how it behaves on your project?

File: src/Sulu/Component/CustomUrl/Manager/CustomUrlManager.php [hunk 1]
```diff
@@ -192,9 +192,15 @@ public function findByUrl($url, $webspaceKey, $locale = null)
 
     public function findByPage(UuidBehavior $page)
     {
+        $descendentPaths = [];
+        foreach ($this->webspaceManager->getWebspaceCollection() as $webspace) {
+            $descendentPaths[] = sprintf('ISDESCENDANTNODE(a, "/cmf/%s/custom-urls")', $webspace->getKey());
+        }
```
Comment (alexander-schranz) [inline src/Sulu/Component/CustomUrl/Manager/CustomUrlManager.php]: Can we not just get the webspace from the page? I don't think that cross webspace custom urls are possible.

Comment (wachterjohannes) [reply src/Sulu/Component/CustomUrl/Manager/CustomUrlManager.php]: could try that if @rgericke-elb can approve the performance improvement

Comment (rgericke-elb) [reply src/Sulu/Component/CustomUrl/Manager/CustomUrlManager.php]: seems to be doing its job. publishing and deleting is much faster with this fix

**Review by alexander-schranz — APPROVED** [review]

**Review by alexander-schranz — CHANGES_REQUESTED** [review]
See: https://github.com/sulu/sulu/pull/7014/files#r1109525831

File: src/Sulu/Component/CustomUrl/Manager/CustomUrlManager.php [hunk 2]
```diff
@@ -192,9 +194,22 @@ public function findByUrl($url, $webspaceKey, $locale = null)
 
     public function findByPage(UuidBehavior $page)
     {
+        $webspaceKeys = array_map(function (Webspace $webspace) {
+            return $webspace->getKey();
+        }, $this->webspaceManager->getWebspaceCollection()->toArray());
+        if ($page instanceof WebspaceBehavior) {
+            $webspaceKeys = [$page->getWebspaceName()];
```
Comment (wachterjohannes) [inline src/Sulu/Component/CustomUrl/Manager/CustomUrlManager.php:201]: @alexander-schranz it triggers me but `webspaceName` is the `webspaceKey` 😂

**Comment by wachterjohannes** [comment]
@alexander-schranz is fixed and ready to merge

**Review by alexander-schranz — APPROVED** [review]

**Pull request merged**